docs/DataRequirements.md

```diff
@@ -0,0 +1,252 @@
+
+## IBC modules RPC
+
+- Queries client, connection and channel-associated data and states.
+- Used to retrieve commitment proofs for every IBC message relayed
+- To obtain the client upgraded state, while the relayer is handling chain
+  upgrades.
+
+- Status: RPC methods Implemented expect for the upgraded client state query.
+
+### Channel endpoints
+
+- `ibc_channelClientState`: Requests the client state associated with a
+  specified channel.
+- `ibc_channels`: Requests all of the channels associated with the chain.
+- `ibc_channelConnections`: Requests the connection associated with a specified
+  channel.
+- `ibc_nextSequenceReceive`: Requests the sequence number of the next receive
+  packet for a specified channel.
+- `ibc_packetCommitments`: Requests the packet commitments associated with a
+  specified channel.
+- `ibc_packetAcknowledgements`: Requests the packet acknowledgments associated
+  with a specified channel. -`ibc_unreceivedAcks`: Requests the unreceived
+acknowledgments associated with a specified channel. -`ibc_unreceivedPackets`:
+Requests the unreceived packet sequences associated with a specified channel.
+
+> _*Note:*_ The `PacketAcknowledgements`, `UnreceivedAcks`, and
+> `UnreceivedPackets` queries each accept a vector of `sequences` in order to
+> specify which packet commitments to fetch acknowledgements for. In the case
+> where an empty vector is passed, the queries will simply return all
+> acknowledgements for all outstanding packet commitments.
+
+### Client endpoints
+
+- `ibc_clientStates`: Requests all client states associated with the chain.
+- `ibc_ConsensusStates`: Requests all the consensus states associated with a
+  specified client.
+- `ibc_consensusStateHeights`: Requests all the consensus state heights
+  associated with a specified client.
+- `ibc_upgradedClientState`: Requests the upgraded client state associated with a
+  specified client.
+
+### Connection endpoints
+
+- `ibc_clientConnections`: Requests all connections associated with a specified client.
+- ‍`ibc_connections`: Requests all connections associated with the chain.
+
+## Rollup WebSocket
+
+### `/subscribe_aggregated_proofs`
+
+- Subscribe to the rollup node's websocket and listen to aggregated proofs every
+  time a proof is written for the rollup.
+
+- Status: Nothing available yet.
+
+### `/subscribe_events`
+
+- (Nice to have) Connect to the rollup node's websocket and subscribes to the
+  ibc events.
+
+- Status: The `ledger_subscribeSlots` endpoint provides a stream of
+  [`SlotResponse`](https://github.com/Sovereign-Labs/sovereign-sdk/blob/bd469c70fc1227a7785fb177a34de21bb6d5eb08/rollup-interface/src/node/rpc/mod.rs#L136-L149)
+  type, encompassing the processed batches. Within this, we gain access to a
+  list of transactions and so on related events. However, for obtaining IBC
+  events, it is necessary to implement a filter on the SlotResponse type to
+  exclude non-IBC batches, txs and events.
```

Comment:
I am more inclined to not use this but rather just the aggregate proof notification.
If we cleared packets up to some `previous_height` and we get an aggregate proof for `next_height` then we can query all events that were emitted in between and relay those.

Reply:
hmmm...actually thinking about this more. Not sure about the avail of the websocket on the rollup but, if it is there, we could batch all IBC events that occurred in `next_height` (like we do today in hermes) and use the `post_state_root` in the proof notification as the commitment root for all of the IBC events. And prepend the client update with the celestia header that included the proof and the proof itself.

Reply:
Looked at the code and it seems that there is no websocket/ notification interface and would need to use the RPC you mentioned. should be fine.